addons/udes_stock_refactoring/tests/test_refactoring.py

```diff
@@ -1078,3 +1078,160 @@ def test_split_mixed_products_partial_reserve(self):
         move_format = [(m.product_id, m.product_uom_qty, m.reserved_availability) for m in moves]
         expected_move_format = [(self.apple, 2, 0), (self.banana, 3, 0)]
         self.assertCountEqual(move_format, expected_move_format)
+
+
+class RefactorByPickingZoneTestCase(TestRefactoringBase):
+    """Tests for refactoring by picking zone."""
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        cls.Pick = cls.env["stock.picking"]
+
+        # Find the picking types and locations we need.
+        cls.picking_type_pick.write(
+            {
+                "u_post_assign_action": "group_by_move_line_key",
+                "u_move_line_key_format": "{move_id.picking_id.origin},{location_id.u_picking_zone_id}",
+            }
+        )
+
+        zone1 = cls.create_location(
+            "zone1", location_id=cls.stock_location.id, usage="view", u_is_picking_zone=True
+        )
+        zone2 = cls.create_location(
+            "zone2", location_id=cls.stock_location.id, usage="view", u_is_picking_zone=True
+        )
+        cls.zones = zone1 | zone2
+
+        cls.loc1 = cls.create_location("loc1", location_id=zone1.id)
+        cls.loc2 = cls.create_location("loc2", location_id=zone1.id)
+        cls.loc3 = cls.create_location("loc3", location_id=zone2.id)
+
+        cls.create_quant(cls.apple.id, cls.loc1.id, 2, cls.create_package().id)
+        cls.create_quant(cls.banana.id, cls.loc2.id, 2, cls.create_package().id)
+        cls.create_quant(cls.cherry.id, cls.loc3.id, 2, cls.create_package().id)
+
+    def test_refactors_by_picking_zone(self):
+        """The system will refactor by picking zone."""
+        origin = "TEST_ORIGIN"
+        # One picking, stock in each zone.
+        products_info = [{"product": self.apple, "qty": 1}, {"product": self.cherry, "qty": 1}]
+        original_picking = self.create_picking(
+            self.picking_type_pick, products_info=products_info, origin=origin, confirm=True
+        )
+
+        original_picking.action_assign()
```

Comment:
You should be able to pass in `assign=True` to `create_picking()` and it should product an assigned picking

Reply:
Ah I was asserting the length of `original_picking` at first, but then removed as I don't like asserting pre-conditions.  I'll change.